File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> points=new HashMap<>();
        for(int i=0;i<name.length;i++){
            points.put(name[i],yearning[i]);
        }
        return Arrays.stream(photo)
            .mapToInt(p->
                Arrays.stream(p)
                    .mapToInt(m->
                        points.getOrDefault(m,0)
                    ).sum()
            ).toArray();
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const yearningByName = new Map();
    for(let i = 0; i < name.length; i++)
        yearningByName.set(name[i],yearning[i]);
    return photo.map((arr)=>
                     arr.map((p)=>yearningByName.get(p) || 0)
                     .reduce((a,b)=>a+b));
}